crates/stages/src/stages/bodies.rs

```diff
@@ -572,24 +659,36 @@ mod tests {
             fn seed_execution(&mut self, input: ExecInput) -> Result<Self::Seed, TestRunnerError> {
                 let start = input.checkpoint().block_number;
                 let end = input.target();
+
+                let snapshot_provider = self.db.factory.snapshot_provider().expect("should exist");
+
                 let mut rng = generators::rng();
                 let blocks = random_block_range(&mut rng, start..=end, GENESIS_HASH, 0..2);
                 self.db.insert_headers_with_td(blocks.iter().map(|block| &block.header))?;
                 if let Some(progress) = blocks.first() {
                     // Insert last progress data
-                    self.db.commit(|tx| {
+                    {
+                        let tx = self.db.factory.provider_rw()?.into_tx();
+                        let mut snapshotter =
+                            snapshot_provider.writer(start, SnapshotSegment::Transactions)?;
+
                         let body = StoredBlockBodyIndices {
-                            first_tx_num: 0,
+                            first_tx_num: 1,
                             tx_count: progress.body.len() as u64,
                         };
+
+                        for _ in 0..progress.number {
+                            snapshotter.increment_block(SnapshotSegment::Transactions)?;
+                        }
```

Comment:
Maybe we just override the end of the range here instead? Since each call to `increment_block` creates a new `RangeInclusive` on the heap so it's prob not too cheap (minor perf)